db/migrate/20140522185700_change_templatekind_to_bootdisk.rb

```diff
@@ -1,14 +1,22 @@
 # frozen_string_literal: true
 
 class ChangeTemplatekindToBootdisk < ActiveRecord::Migration[4.2]
+  class FakeConfigTemplate < ApplicationRecord
+    self.table_name = if ActiveRecord::Base.connection.migration_context.get_all_versions.include?(20_150_514_072_626)
+                        'templates'
+                      else
+                        'config_templates'
+                      end
```

Comment:
I think this is wrong.

https://github.com/theforeman/foreman/commit/9851420902165fc23d0ea1f6d53922a5f6a37c6a dropped the old `20150514114044_migrate_ptables_to_templates.rb` migration, so the `include?` will be `false`, and you'd end up with `config_templates` name, not `templaes` for the table name.

Reply:
Hmm, no, it dropped only parts of that migration. So it actually could work.

@m-bucher could you have a look, was the removal in #110 purely cleanup, or was there a bug?

Reply:
IIRC we had problems during migration in certain cases. Though I am not sure in which scenarios specifically :thinking:

Reply:
Hmm but the statement queries `db/migrate/20150514072626_add_type_to_config_template.rb` and not `20150514114044_migrate_ptables_to_templates.rb`. The former migration is still present even today in develop.

Reply:
Yeah, I did not notice the conversation after the initial comment...

Anyway, isn't safer to simply test for a class constant presence? Something like `if defined? ConfigTemplates` ?

Reply:
I pushed another commit that feels safer to me, what you guys think? That does not depend on any migration numbers or whatever, it simply tests if older class constant exists, if it does not it is named the new way.

Reply:
> Hmm but the statement queries `db/migrate/20150514072626_add_type_to_config_template.rb` and not `20150514114044_migrate_ptables_to_templates.rb`. The former migration is still present even today in develop.

Heh. I failed on Ctrl+F and this silly notation of the migration version. You're right.